data-prepper-logstash-configuration/src/main/java/org/opensearch/dataprepper/logstash/mapping/AbstractLogstashPluginAttributesMapper.java

```diff
@@ -36,15 +36,22 @@
                 .stream()
                 .filter(logstashAttribute -> !customMappedAttributeNames.contains(logstashAttribute.getAttributeName()))
                 .forEach(logstashAttribute -> {
-            if (mappedAttributeNames.containsKey(logstashAttribute.getAttributeName())) {
-                pluginSettings.put(
-                        mappedAttributeNames.get(logstashAttribute.getAttributeName()),
-                        logstashAttribute.getAttributeValue().getValue()
-                );
-            }
-            else {
-                LOG.warn("Attribute name {} is not found in mapping file.", logstashAttribute.getAttributeName());
-            }
+                    final String logstashAttributeName = logstashAttribute.getAttributeName();
+                    final String dataPrepperAttributeName = mappedAttributeNames.get(logstashAttributeName);
+
+                    if (mappedAttributeNames.containsKey(logstashAttributeName)) {
+                        if (dataPrepperAttributeName.startsWith("!") && logstashAttribute.getAttributeValue().getValue() instanceof Boolean) {
```

Comment:
Should we warn users they are using the ! operator on a non-boolean value? In my opinion this would be the equivalent of giving a bad config and could have unexpected results.

Reply:
I didn't get it completely. Do you mean that we if someone tries to use "!" on non-boolean attribute we need to warn them?

Reply:
Warn them or fail to parse the config. Similar to if I used this config yaml:
```
ssl: false
invalid-key: Hello
```

DataPrepper would throw the exception:
```
Exception in thread "main" java.lang.IllegalArgumentException: Invalid DataPrepper configuration file.
```